File: slack-poster/src/test/java/dev/number6/slackposter/generate/MapAttributeValueGenerator.java
package dev.number6.slackposter.generate;

import uk.org.fyodor.generators.Generator;
import com.amazonaws.services.dynamodbv2.model.AttributeValue;

import static uk.org.fyodor.generators.RDG.map;

public class MapAttributeValueGenerator implements Generator<AttributeValue>{

    private final Generator<String> key;
    private final Generator<AttributeValue> val;
    private final int size;

    public MapAttributeValueGenerator(Generator<String> key, Generator<AttributeValue> val, int size) {
        this.key = key;
        this.val = val;
        this.size = size;
    }

    @Override
    public AttributeValue next() {
        return new AttributeValue().withM(map(key, val, size).next());
    }
}

File: slack-poster/src/test/kotlin/dev/number6/slackposter/generate/MapAttributeValueGenerator.kt
package dev.number6.slackposter.generate

import com.amazonaws.services.dynamodbv2.model.AttributeValue
import uk.org.fyodor.generators.Generator
import uk.org.fyodor.generators.RDG

class MapAttributeValueGenerator(private val key: Generator<String>, private val `val`: Generator<AttributeValue>, private val size: Int) : Generator<AttributeValue> {
    override fun next(): AttributeValue {
        return AttributeValue().withM(RDG.map(key, `val`, size).next())
    }

}